src/lib.rs

```diff
@@ -89,7 +89,7 @@ where
     enum Unreachable {}
 
     try_array_init(
-        // monomorphise into an unfailible version
+        // monomorphise into an unfallible version
```

Comment:
```suggestion
        // monomorphise into an infallible version
```

No, you were correct in that "unfallible" is not a word, your fix was incorrect: it should be "infallible"